File: AMR12D.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
  int i,j,t,len;
  string s;
  cin>>t;
  while(t--){
    cin>>s;
    len = s.length();

    for(i=0;i<len;i++)
      if(!(s[i]==s[len-1-i]))break;
    cout << (i==len ? "YES" : "NO") << endl;
  }

  return 0;
}

// AC

File: AMR12D.py
#!/usr/bin/env python


# brain strain .... no mood


if __name__=='__main__':
    s = raw_input()
    l = len(s)
    print "YES" if s[0:l/2]==reversed(s[(l/2)+1:l+1]) else "NO"
    print s[0:l/2]
    print s[(l/2):l+1]
        
